tests/ui/mir/static_fnptr.rs

```diff
@@ -0,0 +1,21 @@
+//! Verify that we correctly handle fn pointer provenance in MIR optimizations.
+//! By asking to inline `static_fnptr::bar`, we have two copies of `static_fnptr::foo`, one in the
+//! auxiliary crate and one in the local crate CGU.
```

Comment:
This seems to rely on `foo` being automatically marked as `inline`? Does `inline(never)` on `foo` make the issue go away?

If yes it'd be better to mark it as `inline` explicitly to avoid rely on the auto-inline heuristic.